static-site/src/util/blogPosts.js

```diff
@@ -2,7 +2,9 @@ import fs from 'fs';
 import path from 'path';
 import { fileURLToPath } from 'url';
 import matter from 'gray-matter';
-import { startCase } from "lodash"
+import lodash from 'lodash';
+
+const { startCase } = lodash;
```

Comment:
`import { startCase } from "lodash"` often allows tree-shaking (cf. `import lodash from 'lodash';`). I can't remember the details of what nextjs is doing here / what we've asked it to do. Might be worth [analysing the bundle](https://nextjs.org/docs/pages/building-your-application/optimizing/package-bundling#analyzing-javascript-bundles) here to check.

Reply:
> `import { startCase } from "lodash"` often allows tree-shaking (cf. `import lodash from 'lodash';`). I can't remember the details of what nextjs is doing here / what we've asked it to do. 

When this file contains `import { startCase } from 'lodash';`, running the feed generation script produces this error:

```javascript
> npm run build:feeds                                                                                                                 v20.9.0

> build:feeds
> node scripts/generate-blog-feeds.js

file:///Users/genehack/nextstrain/nextstrain.org/static-site/src/util/blogPosts.js:9
import { startCase } from 'lodash';
         ^^^^^^^^^
SyntaxError: Named export 'startCase' not found. The requested module 'lodash' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'lodash';
const { startCase } = pkg;

    at ModuleJob._instantiate (node:internal/modules/esm/module_job:131:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:213:5)
    at async ModuleLoader.import (node:internal/modules/esm/loader:316:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:66:12)

Node.js v20.9.0
```

(Trying to evaluate the `blogPosts.js` file directly, in the `static-site` directory, generates the exact same error.)

I'm not sure why this error doesn't happen when the file is included in the 2 JSX files that consume it. 

> Might be worth [analysing the bundle](https://nextjs.org/docs/pages/building-your-application/optimizing/package-bundling#analyzing-javascript-bundles) here to check.

The only place this file gets included in the site is in 2 JSX files that are used to produce static site files — I don't think the `lodash` Javascript would need to end up in the bundle? But maybe I'm confused about how the "static" file generation works in the production build.

Reply:
> The only place this file gets included in the site is in 2 JSX files that are used to produce static site files — I don't think the lodash Javascript would need to end up in the bundle? But maybe I'm confused about how the "static" file generation works in the production build.

In this case it's not ending up in the bundle - next.js is (correctly) not including it even tho the files which import it themselves export frontend JSX.

> Might be worth [analysing the bundle](https://nextjs.org/docs/pages/building-your-application/optimizing/package-bundling#analyzing-javascript-bundles) here to check.

I did this to double check and it identified some other issues. Will make an issue.

Reply:
> I did this to double check and it identified some other issues. Will make an issue.

Cancel that, was looking at stat sizes not parsed/compressed sizes.